main.go

```diff
@@ -114,60 +112,99 @@ func main() {
 
 	eventLoopContext, eventLoopCancel := context.WithCancel(context.Background())
 
+	// Validate this service against Zebedee
 	serviceIdentityValidated := make(chan bool)
 	go func() {
 		if err = checkServiceIdentity(eventLoopContext, cfg.ZebedeeURL, cfg.ServiceAuthToken); err != nil {
 			log.ErrorC("could not obtain valid service account", err, nil)
-			eventLoopDone <- true
 		}
 		serviceIdentityValidated <- true
 	}()
 
-	eventConsumer.Start(eventLoopContext, eventLoopDone, serviceIdentityValidated)
+	// Start Event Consumer (only if it is available)
+	if serviceList.Consumer {
+		eventConsumer.Start(eventLoopContext, eventLoopDone, serviceIdentityValidated)
+	}
+
+	// Log non-fatal errors, without exiting
+	go func() {
+		var consumerErrors chan (error)
+		if serviceList.Consumer {
+			consumerErrors = syncConsumerGroup.Errors()
+		} else {
+			consumerErrors = make(chan error, 1)
+		}
+
+		var producerErrors chan (error)
+		if serviceList.DimensionExtractedProducer {
+			producerErrors = service.DimensionExtractedProducer.Errors()
+		} else {
+			producerErrors = make(chan error, 1)
+		}
+
+		select {
+		case consumerError := <-consumerErrors:
+			log.ErrorC("kafka consumer", consumerError, nil)
+		case producerError := <-producerErrors:
+			log.ErrorC("kafka producer", producerError, nil)
+		case apiError := <-apiErrors:
+			log.ErrorC("server error", apiError, nil)
+		}
+	}()
 
-	// block until a fatal error, signal or eventLoopDone - then proceed to shutdown
+	// Block until a fatal error occurs
 	select {
 	case <-eventLoopDone:
```

Comment:
Should we not remove this case to prevent the app proceeding to shutdown early?

Reply:
Done